### PR TITLE
use GSO

### LINKED ISSUE #2877: Boosting performance: UDP Offloading and GSO
Hello, I've been instrumenting numerous QUIC implementations for performance testing.

In particular, I see [`quiche`] from CloudFlare implemented UDP offloading and GSO earlier this year. [[link]](https://blog.cloudflare.com/accelerating-udp-packet-transmission-for-quic/)
`quic-go` has given me 470.3 Mbps with loopback setting, so it seems to have no support for both accelerations yet. (I briefly looked up the code)

Would anyone confirm this for `quic-go`?
Maybe we could keep open this issue for future implementation, too.

### PR DESCRIPTION
**Pull request by marten-seemann**
Fixes #2877. Depends on #3806.

### CODE REVIEW

**Review by lucas-clemente — APPROVED** [review]

File: sys_conn_df_linux.go [hunk 1]
```diff
@@ -34,7 +41,36 @@ func setDF(rawConn syscall.RawConn) (bool, error) {
...
 
+func maybeSetGSO(rawConn syscall.RawConn) bool {
+	var setErr error
+	if err := rawConn.Control(func(fd uintptr) {
+		setErr = unix.SetsockoptInt(int(fd), syscall.IPPROTO_UDP, UDP_SEGMENT, 1)
+	}); err != nil {
+		setErr = err
+	}
+	if setErr != nil {
+		log.Println("failed to enable GSO")
```
[the remark this answers is not in the document]
Comment (marten-seemann) [reply sys_conn_df_linux.go:59]: Missed this comment. Will address in a follow-up PR.

**Comment by bt90** [comment]
As we're sending a lot more efficient now, we should check if we would profit from a larger send buffer.

**Comment by otbutz** [comment]
Any ETA for the v0.36 release? It would be really nice to have GSO in the upcoming caddy release.

**Comment by otbutz** [comment]
caddy v2.7.0 is currently in the beta phase: https://github.com/caddyserver/caddy/releases/tag/v2.7.0-beta.1

> What’s the timeline for that release?

Pinging @mholt and @francislavoie as I don't have an answer and they are the ones who make the decision.

I can only argue from the user perspective. Caddy uses HTTP/3 by default and it would be really unfortunate if the next version would be shipped without GSO because the schedules differ by a few days.

**Comment by marten-seemann** [comment]
We're going to release v0.36 today or tomorrow. @mholt What's your policy regarding minor version updates after you've shipped a beta? Would you still be able to include this version in v2.7.0 or is it considered too risky?

**Comment by francislavoie** [comment]
We're planning a second beta, so we can certainly get this in.

**Comment by marten-seemann** [comment]
https://github.com/caddyserver/caddy/pull/5584

**Pull request merged**